### PR TITLE
samples: doc: Added README for RTIO sample

### PR DESCRIPTION
**Pull request by michalpiekos**
Added README.rst for sensor_batch_processing sample. Added rtio.rst listing samples from RTIO subsystem.
Helps fix https://github.com/zephyrproject-rtos/zephyr/issues/27805

### CODE REVIEW

File: samples/subsys/rtio/sensor_batch_processing/README.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
+.. zephyr:code-sample:: sensor_batch_processing
+   :name: Sensor batch processing
+   :relevant-api: rtio
+
+   Implements sensor device using RTIO for asynchronuous data processing.
```
Comment (kartben) [inline samples/subsys/rtio/sensor_batch_processing/README.rst:5]: a typo + changing wording to follow same style as other samples (https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/templates/sample.tmpl#L5-L6)

```suggestion
   Implement a sensor device using RTIO for asynchronous data processing.
```

File: samples/subsys/rtio/sensor_batch_processing/README.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Overview
+********
+
+This sample application demonstrates the use of the :ref:`rtio` library
+which is a framework for doing asynchronuous operation chains.
```
Comment (kartben) [inline samples/subsys/rtio/sensor_batch_processing/README.rst:11]: same typo :) + suggested wording improvement
```suggestion
This sample application demonstrates the use of the :ref:`rtio` framework for
doing asynchronuous operation chains.
```

File: samples/subsys/rtio/sensor_batch_processing/README.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+application for the native_sim target.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/rtio
+   :board: native_sim
+   :goals: build run
+   :compact:
+
+When running the output on the console shows the operations of
+submitting for, consuming and processing the sensor data.
```
Comment (kartben) [inline samples/subsys/rtio/sensor_batch_processing/README.rst:35]:
```suggestion
When running, the output on the console shows the operations of
submitting for, consuming, and processing the sensor data.
```

File: samples/subsys/rtio/rtio.rst [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.. zephyr:code-sample-category:: rtio
+   :name: Real Time I/O
```
Comment (kartben) [inline samples/subsys/rtio/rtio.rst:2]:
```suggestion
   :name: Real Time I/O (RTIO)
```

**Review by kartben — CHANGES_REQUESTED** [review]
thanks for this! some minor typos or grammar corrections

File: samples/subsys/rtio/sensor_batch_processing/README.rst (hunk 1 above)
Comment (michalpiekos) [reply samples/subsys/rtio/sensor_batch_processing/README.rst]: Fixed

File: samples/subsys/rtio/sensor_batch_processing/README.rst (hunk 2 above)
Comment (michalpiekos) [reply samples/subsys/rtio/sensor_batch_processing/README.rst]: Fixed

File: samples/subsys/rtio/sensor_batch_processing/README.rst (hunk 3 above)
Comment (michalpiekos) [reply samples/subsys/rtio/sensor_batch_processing/README.rst]: Fixed

File: samples/subsys/rtio/rtio.rst (hunk 4 above)
Comment (michalpiekos) [reply samples/subsys/rtio/rtio.rst]: Fixed

**Review by kartben — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Pull request merged**